app/analyzers/beaconing.py

```diff
@@ -0,0 +1,139 @@
+
+                if len(outliers) > 0:
+                    unique_summaries = len(set(o.outlier_dict["summary"] for o in outliers))
+                    logging.logger.info("total outliers in batch processed: " + str(len(outliers)) + " [" + str(unique_summaries) + " unique summaries]")
+                    outlier_batches_trend += 1
+                else:
+                    logging.logger.info("no outliers detected in batch")
+                    outlier_batches_trend -= 1
+
+                # Reset data structures for next batch
+                eval_terms_array = defaultdict()
+                total_terms_added = 0
+
+        self.print_analysis_summary()
+
+    def extract_additional_model_settings(self):
+        self.model_settings["target"] = settings.config.get(self.config_section_name, "target").replace(' ', '').split(",")  # remove unnecessary whitespace, split fields
+        self.model_settings["aggregator"] = settings.config.get(self.config_section_name, "aggregator").replace(' ', '').split(",")  # remove unnecessary whitespace, split fields
+        self.model_settings["trigger_sensitivity"] = settings.config.getint(self.config_section_name, "trigger_sensitivity")
+        self.model_settings["batch_eval_size"] = settings.config.getint("beaconing", "beaconing_batch_eval_size")
+
+    @staticmethod
+    def add_term_to_batch(eval_terms_array, aggregator_value, target_value, observations, doc):
+        if aggregator_value not in eval_terms_array.keys():
+            eval_terms_array[aggregator_value] = defaultdict(list)
+
+        eval_terms_array[aggregator_value]["targets"].append(target_value)
+        eval_terms_array[aggregator_value]["observations"].append(observations)
+        eval_terms_array[aggregator_value]["raw_docs"].append(doc)
+
+        return eval_terms_array
+
+    def evaluate_batch_for_outliers(self, terms=None):
+        # Initialize
+        outliers = list()
+
+        # In case we want to count terms within an aggregator, it's a bit easier.
+        # For example:
+        # terms["smsc.exe"][A, B, C, D, D, E]
+        # terms["abc.exe"][A, A, B]
+        # is converted into:
+        # First iteration: "smsc.exe" -> counted_target_values: {A: 1, B: 1, C: 1, D: 2, E: 1}
+        # For each aggregator, we iterate over all terms within it:
+        # term_value_count for a document with term "A" then becomes "1" in the example above.
+        # we then flag an outlier if that "1" is an outlier in the array ["1 1 1 2 1"]
+        for _, aggregator_value in enumerate(terms):
+            # Count percentage of each target value occuring
+            counted_targets = Counter(terms[aggregator_value]["targets"])
+            counted_target_values = list(counted_targets.values())
+
+            logging.logger.debug("terms count for aggregator value " + aggregator_value + " -> " + str(counted_targets))
+
+            if len(counted_targets) < 10:
```

Comment:
This 10 seems a bit of an arbitrary value, I advise to at least move it to a constant at the start of the module or something, maybe even make the cutoff value configurable.